### PR TITLE
Fixes Issues after testing plugin with WP 5.7 RC release

### PR DESCRIPTION
**Pull request by tremidkhar**
# Fix for #1726 

Issue where the `wp_localize_script()` is being called incorrectly. The `wp_localize_script()` function required an associate array as the third argument $l10n.

### CODE REVIEW

File: app/assets/js/rtMedia.backbone.js [hunk 1]
```diff
@@ -1541,7 +1539,7 @@ jQuery( document ).ready( function( $ ) {
 		} );
 
 		objUploadView.uploader.bind( 'UploadProgress', function( up, file ) {
-			//$( "#" + file.id + " .plupload_file_status" ).html( rtmedia_uploading_msg + '( ' + file.percent + '% )' );
+			//$( "#" + file.id + " .plupload_file_status" ).html( rtmedia_main.rtmedia_uploading_msg + '( ' + file.percent + '% )' );
```
Comment (SID177) [inline app/assets/js/rtMedia.backbone.js:1542]: Remove this comment

File: app/main/RTMedia.php [hunk 2]
```diff
@@ -1214,6 +1214,11 @@ public function create_table_error_notice() {
 	public function enqueue_scripts_styles() {
 		global $rtmedia, $bp, $rtmedia_interaction;
 
+		$rtmedia_main = array();
+		$rtmedia_magnific = array();
+		$rtmedia_backbone = array();
+		$rtmedia_bp_tpl = array();
```
Comment (SID177) [inline app/main/RTMedia.php:1220]: Check phpcs here

File: app/main/RTMedia.php [hunk 3]
```diff
@@ -1274,7 +1279,7 @@ public function enqueue_scripts_styles() {
 				true
 			);
 			// localize for rtmedia js.
-			wp_localize_script( 'rtmedia-main', 'bp_template_pack', $bp_template );
+			wp_localize_script( 'rtmedia-main', 'rtmedia_bp', array( 'template_pack' => $bp_template ) );
```
Comment (SID177) [inline app/main/RTMedia.php:1282]: phpcs

File: app/main/RTMedia.php [hunk 4]
```diff
@@ -1298,63 +1303,64 @@ public function enqueue_scripts_styles() {
...
-		wp_localize_script( 'rtmedia-main', 'rtmedia_ajax_url', admin_url( 'admin-ajax.php' ) );
-		wp_localize_script( 'rtmedia-main', 'rtmedia_media_slug', RTMEDIA_MEDIA_SLUG );
-		wp_localize_script( 'rtmedia-main', 'rtmedia_lightbox_enabled', strval( $this->options['general_enableLightbox'] ) );
+		$rtmedia_main['media_delete_confirmation'] = $media_delete_confirmation;
+		$rtmedia_main['rtmedia_ajaxurl']           = admin_url( 'admin-ajax.php' );
+		$rtmedia_main['media_delete_success']      = $media_delete_success;
+
+		$rtmedia_main['rtmedia_ajax_url'] = admin_url( 'admin-ajax.php' );
+		$rtmedia_main['rtmedia_media_slug'] = RTMEDIA_MEDIA_SLUG;
+		$rtmedia_main['rtmedia_lightbox_enabled'] = strval( $this->options['general_enableLightbox'] );
```
Comment (SID177) [inline app/main/RTMedia.php:1313]: phpcs

File: app/main/RTMediaUploadTerms.php [hunk 5]
```diff
@@ -115,7 +115,7 @@ public function enqueue_scripts_styles() {
 					)
 				);
 
-				wp_localize_script( 'rtmedia-main', 'rtmedia_upload_terms_check_terms_message', $general_upload_terms_error_message );
+				wp_localize_script( 'rtmedia-main', 'rtmedia_upload_terms', array( 'check_terms_message' => $general_upload_terms_error_message ) );
```
Comment (SID177) [inline app/main/RTMediaUploadTerms.php:118]: phpcs

File: app/main/controllers/activity/RTMediaBuddyPressActivity.php [hunk 6]
```diff
@@ -785,8 +785,13 @@ public function bp_after_activity_post_form() {
 			wp_enqueue_script( 'rtmedia-backbone', false, '', RTMEDIA_VERSION, true );
 			$is_album        = is_rtmedia_album() ? true : false;
 			$is_edit_allowed = is_rtmedia_edit_allowed() ? true : false;
-			wp_localize_script( 'rtmedia-backbone', 'is_album', $is_album );
-			wp_localize_script( 'rtmedia-backbone', 'is_edit_allowed', $is_edit_allowed );
+
+			$activity = array(
+				'is_album' => $is_album,
+				'is_edit_allowed' => $is_edit_allowed,
+			);
```
Comment (SID177) [inline app/main/controllers/activity/RTMediaBuddyPressActivity.php:792]: phpcs

**Review by SID177 — CHANGES_REQUESTED** [review]

File: app/main/RTMediaUploadTerms.php (hunk 5 above)
Comment (tremidkhar) [reply app/main/RTMediaUploadTerms.php:118]: Cannot reproduce the PHPCS error.

File: app/main/RTMedia.php (hunk 3 above)
Comment (tremidkhar) [reply app/main/RTMedia.php:1282]: Cannot reproduce the PHPCS error.

File: app/admin/RTMediaAdmin.php [hunk 7]
```diff
@@ -963,10 +956,21 @@ public function ui( $hook ) {
...
+					array(
+						'on_label'                     => __( 'ON', 'buddypress-media' ),
+						'off_label'                    => __( 'OFF', 'buddypress-media' ),
+						'admin_ajax'                   => $admin_ajax,
+						'admin_url'                    => admin_url(),
+						'fileupload_url'               => RTMEDIA_URL . 'app/helper/rtUploadAttachment.php', /* path for file upload using ajax */
+						'settings_url'                 => esc_url( add_query_arg( array( 'page' => 'rtmedia-settings' ), ( is_multisite() ? network_admin_url( 'admin.php' ) : admin_url( 'admin.php' ) ) ) ) . '#privacy_enabled',
+						'settings_rt_album_import_url' => esc_url( add_query_arg( array( 'page' => 'rtmedia-settings' ), ( is_multisite() ? network_admin_url( 'admin.php' ) : admin_url( 'admin.php' ) ) ) ),
+					)
+				);
```
Comment (SID177) [inline app/admin/RTMediaAdmin.php:973]: I don't see any changes made in settings script and so it's giving console errors.

Also if the var names has changed here, does it work with all add-ons? If it doesn't work please do the same as it's done for frontend scripts.

File: app/assets/js/rtMedia.backbone.js [hunk 8]
```diff
@@ -8,6 +8,31 @@ var rtmedia_load_template_flag = true;
 var rtmedia_add_media_button_post_update = false;
 
 
+jQuery( document ).ready( function () {
+
+	// Need to pass the object[key] as global variable.
+	if ( 'object' === typeof rtmedia_backbone ) {
+		for( var key in rtmedia_backbone ) {
```
Comment (SID177) [inline app/assets/js/rtMedia.backbone.js:15]:
```suggestion
		for ( var key in rtmedia_backbone ) {
```

**Review by SID177 — CHANGES_REQUESTED** [review]

**Pull request merged**